File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s is None or t is None:
            return False
        i=0
        for c in t:
            if i<len(s) and s[i] == c:
                i+=1
                
        return i == len(s)

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s == null || t == null) return false;
        int i=0;       

        for(char c : t.toCharArray()){
            if (i< s.length() && s.charAt(i) == c){
                i++;
            }
        }

        return i == s.length();
    }
}